src/main/java/org/opensearch/knn/index/KNNIndexCache.java

```diff
@@ -312,38 +310,122 @@ public KNNIndexCacheEntry loadIndex(String indexPathUrl, String indexName) throw
         // the entry
         fileWatcher.init();
 
-        final KNNIndex knnIndex = KNNIndex.loadIndex(indexPathUrl, getQueryParams(indexName),
-                SpaceType.getSpace(KNNSettings.getSpaceType(indexName)).getValue());
+        // loadIndex from different library
+        String engineName;
+        Map<String, Object> parameters;
+        if (indexPathUrl.endsWith(KNNEngine.NMSLIB.getExtension())
+                || indexPathUrl.endsWith(KNNEngine.NMSLIB.getCompoundExtension())) {
+            engineName = KNNEngine.NMSLIB.getName();
+            parameters = ImmutableMap.of(
+                    KNNConstants.SPACE_TYPE, spaceType.getValue(),
+                    KNNConstants.HNSW_ALGO_EF_SEARCH, KNNSettings.getEfSearchParam(indexName)
+            );
+        } else {
+            throw new IllegalArgumentException("[KNN] Invalid engine type for path: " + indexPathUrl);
```

Comment:
```indexPathUrl``` should be hidden to the users. May be just have ```indexName``` in the message. We can 
```indexPathUrl``` to a  debug log for debugging.

Reply:
Sure, makes sense.

Reply:
Actually, because this is wrapped in a try catch block, I will just add the indexName in the second exception